dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -600,8 +600,16 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         auto args = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(payload, length);
 
         auto lock{ m_lock.lock_exclusive() };
-        m_foregroundHandlers(*this, args);
-        *foregroundHandled = args.Handled();
+        if (m_foregroundHandlers)
```

Comment:
Do we need a lock_shared to get grab the bool from m_foregroundHandlers?

Reply:
What difference would it make from the current lock_esclusive?

Reply:
Paul is correct. This should be a shared lock. You aren't modifying what m_foreGroundHandlers is pointing to. So shared since it is equivalent to a read operation.